code/game/turfs/closed.dm

```diff
@@ -168,27 +168,40 @@
 
 	if(istype(I, /obj/item/tool/pickaxe/plasmacutter) && !user.do_actions)
 		var/obj/item/tool/pickaxe/plasmacutter/P = I
-		if(!ismineralturf(src) && !istype(src, /turf/closed/gm/dense) && !istype(src, /turf/closed/glass) && !istype(src, /turf/closed/desertdamrockwall) && !istype(src, /turf/closed/brock))
+		if(CHECK_BITFIELD(resistance_flags, RESIST_ALL) || CHECK_BITFIELD(resistance_flags, PLASMACUTTER_IMMUNE))
 			to_chat(user, span_warning("[P] can't cut through this!"))
 			return
-		if(!P.start_cut(user, name, src))
+		if(iswallturf(src)) //walls handle plasma cutter effects on their own
 			return
-
-		if(!do_after(user, PLASMACUTTER_CUT_DELAY, TRUE, src, BUSY_ICON_FRIENDLY))
+		else if(!P.start_cut(user, name, src))
+			return
+		else if(!do_after(user, PLASMACUTTER_CUT_DELAY, TRUE, src, BUSY_ICON_FRIENDLY))
 			return
+		else
+			P.cut_apart(user, name, src) //purely a cosmetic effect
 
-		P.cut_apart(user, name, src)
+		//change targetted turf to a new one to simulate deconstruction
+		if(ismineralturf(src))
+			ChangeTurf(/turf/open/floor/plating/ground/desertdam/cave/inner_cave_floor)
 
-		if(ismineralturf(src) || istype(src, /turf/closed/desertdamrockwall))
+		else if(isdesertrockwallturf(src))
 			ChangeTurf(/turf/open/floor/plating/ground/desertdam/cave/inner_cave_floor)
-		else if(istype(src, /turf/closed/gm/dense))
+
+		else if(isjungleturf(src))
 			ChangeTurf(/turf/open/ground/jungle/clear)
-		else if(istype(src, /turf/closed/brock))
+
+		else if(isiceturf(src) || isicerockturf(src))
+			ChangeTurf(/turf/open/floor/plating/ground/ice)
+
+		else if(isvolcanicturf(src))
```

Comment:
No no no :).

you need to add a var to turfs (let's call it open_turf_type) 
and then you do ChangeTurf(open_turf_type)